From 0c9632ef140a2950b32b9e519fbb822933d78d38 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 16:55:25 -0700
Subject: [PATCH 45/96] HBASE-8118 TestTablePermission depends on the execution order

Reason: Test Failure
Author: Matteo Bertozzi
Ref: CDH-10954
(cherry picked from commit a4a4a853a0da5e79acc8078b56fab975373ead92)
---
 .../security/access/TestTablePermissions.java      |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/security/src/test/java/org/apache/hadoop/hbase/security/access/TestTablePermissions.java b/security/src/test/java/org/apache/hadoop/hbase/security/access/TestTablePermissions.java
index 9c04d59..0410e97 100644
--- a/security/src/test/java/org/apache/hadoop/hbase/security/access/TestTablePermissions.java
+++ b/security/src/test/java/org/apache/hadoop/hbase/security/access/TestTablePermissions.java
@@ -45,6 +45,7 @@ import org.apache.hadoop.hbase.client.Put;
 import org.apache.hadoop.hbase.security.User;
 import org.apache.hadoop.hbase.util.Bytes;
 import org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher;
+import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -104,6 +105,14 @@ public class TestTablePermissions {
     UTIL.shutdownMiniCluster();
   }
 
+  @After
+  public void tearDown() throws Exception {
+    Configuration conf = UTIL.getConfiguration();
+    AccessControlLists.removeTablePermissions(conf, TEST_TABLE);
+    AccessControlLists.removeTablePermissions(conf, TEST_TABLE2);
+    AccessControlLists.removeTablePermissions(conf, AccessControlLists.ACL_TABLE_NAME);
+  }
+
   @Test
   public void testBasicWrite() throws Exception {
     Configuration conf = UTIL.getConfiguration();
-- 
1.7.0.4

